app/permissions_migrations.go

```diff
@@ -74,18 +75,66 @@ const (
 	PermissionManageRemoteClusters           = "manage_remote_clusters" // deprecated; use `manage_secure_connections`
 )
 
-func isRole(roleName string) func(*model.Role, map[string]map[string]bool) bool {
+// Deprecated: This function should only be used if a case arises where team and/or channel scheme roles do not need to be migrated.
+// Otherwise, use isRole.
+func isExactRole(roleName string) func(*model.Role, map[string]map[string]bool) bool {
 	return func(role *model.Role, permissionsMap map[string]map[string]bool) bool {
 		return role.Name == roleName
 	}
 }
 
+// isRole returns true if roleName matches a role's name field or if the a team
+// or channel scheme role matches a "common name". A common name is one of the following role
+// that is common among the system scheme and the team and/or channel schemes:
+//
+// TeamAdmin,
+// TeamUser,
+// TeamGuest,
+// ChannelAdmin,
+// ChannelUser,
+// ChannelGuest,
+// PlaybookAdmin,
+// PlaybookMember,
+// RunAdmin,
+// RunMember
+func isRole(roleName string) func(*model.Role, map[string]map[string]bool) bool {
+	return func(role *model.Role, permissionsMap map[string]map[string]bool) bool {
+		if role.Name == roleName {
+			return true
+		}
+		return isSchemeRoleAssociatedToCommonName(roleName, role)
+	}
+}
+
 func isNotRole(roleName string) func(*model.Role, map[string]map[string]bool) bool {
 	return func(role *model.Role, permissionsMap map[string]map[string]bool) bool {
 		return role.Name != roleName
 	}
 }
 
+func isSchemeRoleAssociatedToCommonName(roleName string, role *model.Role) bool {
+	roleIDToSchemeRoleDisplayName := map[string]string{
+		model.TeamAdminRoleId: sqlstore.SchemeRoleDisplayNameTeamAdmin,
+		model.TeamUserRoleId:  sqlstore.SchemeRoleDisplayNameTeamUser,
+		model.TeamGuestRoleId: sqlstore.SchemeRoleDisplayNameTeamGuest,
+
+		model.ChannelAdminRoleId: sqlstore.SchemeRoleDisplayNameChannelAdmin,
+		model.ChannelUserRoleId:  sqlstore.SchemeRoleDisplayNameChannelUser,
+		model.ChannelGuestRoleId: sqlstore.SchemeRoleDisplayNameChannelGuest,
+
+		model.PlaybookAdminRoleId:  sqlstore.SchemeRoleDisplayNamePlaybookAdmin,
+		model.PlaybookMemberRoleId: sqlstore.SchemeRoleDisplayNamePlaybookMember,
+
+		model.RunAdminRoleId:  sqlstore.SchemeRoleDisplayNameRunAdmin,
+		model.RunMemberRoleId: sqlstore.SchemeRoleDisplayNameRunMember,
+	}
+	displayName, ok := roleIDToSchemeRoleDisplayName[roleName]
+	if !ok {
+		return false
+	}
+	return strings.Contains(role.DisplayName, displayName)
```

Comment:
How robust is this `Contains` against unexpected matches? I guess the `DisplayName` is always under our control? Should we consider just matching the start of the string (if I understand how this `DisplayName` is formed?)

Reply:
It's not future proof if we start allowing schemes role display names to be update by users, but for now we control the `DisplayName` field entirely. 

I've switched from `Contains` to `HasPrefix` for our engineering satisfaction of choosing exactly the right tool.